File: domain/src/main/java/com/rubik/shifttest/domain/usecases/ValidateLastNameUseCase.java
package com.rubik.shifttest.domain.usecases;


public class  ValidateLastNameUseCase implements Validator {
    private static final String LAST_NAME_REGEX = "[a-zA-z]+([ '-][a-zA-Z]+)*";

    @Override
    public boolean execute(CharSequence charSequence) {
        return charSequence.toString().matches(LAST_NAME_REGEX);
    }
}

File: feature/user/registration/src/main/kotlin/ru/rubik/registration/domain/usecases/ValidateLastNameUseCase.kt
package ru.rubik.registration.domain.usecases

import javax.inject.Inject

class ValidateLastNameUseCase @Inject constructor() {
	companion object {

		val LAST_NAME_REGEX = Regex("[a-zA-z]+([ '-][a-zA-Z]+)*")
	}

	operator fun invoke(lastName: String): Boolean = lastName.matches(LAST_NAME_REGEX)
}